jq_service_parser_test.go

```diff
@@ -1,323 +1,60 @@
-                "type": "Progressing"
-            }
-        ],
-        "observedGeneration": 243,
-        "readyReplicas": 2,
-        "replicas": 2,
-        "updatedReplicas": 2
-    }
-}
-`
+//go:embed testdata/deployment.json
+var k8sResource string
+
+//go:embed testdata/simple_config.yaml
+var simpleConfig string
+
+//go:embed testdata/sample_config.yaml
+var sampleConfig string
+
+//go:embed testdata/simple_expectation.json
+var simpleExpectation string
+
+//go:embed testdata/sample_expectation.json
+var sampleExpectation string
 
-func TestJQServiceParserSimpleConfig(t *testing.T) {
-	// Arrange
-	config, err := opslevel_jq_parser.NewServiceRegistrationConfig(opslevel_jq_parser.SimpleConfig)
-	if err != nil {
-		t.Error(err)
+func TestJQServiceParser(t *testing.T) {
+	type TestCase struct {
+		name               string
+		config             string
+		expectedServiceReg string
 	}
-	parser := opslevel_jq_parser.NewJQServiceParser(*config)
-	// Act
-	service, err := parser.Run(k8sResource)
-	if err != nil {
-		t.Error(err)
+	testCases := []TestCase{
+		{"using simple config", simpleConfig, simpleExpectation},
+		{"using sample config", sampleConfig, sampleExpectation},
 	}
-	// Assert
-	autopilot.Equals(t, "web", service.Name)
-	autopilot.Equals(t, "self-hosted", service.Owner)
-	autopilot.Equals(t, "", service.Lifecycle)
-	autopilot.Equals(t, "", service.Tier)
-	autopilot.Equals(t, "", service.Product)
-	autopilot.Equals(t, "", service.Language)
-	autopilot.Equals(t, "", service.Framework)
-	// autopilot.Equals(t, "", service.System)
-	autopilot.Equals(t, 1, len(service.Aliases))
-	autopilot.Equals(t, "k8s:web-self-hosted", service.Aliases[0])
-	autopilot.Equals(t, 1, len(service.TagCreates))
-	autopilot.Equals(t, opslevel.TagInput{Key: "environment", Value: "dev"}, service.TagCreates[0])
-	autopilot.Equals(t, 5, len(service.TagAssigns))
-	autopilot.Equals(t, opslevel.TagInput{Key: "imported", Value: "kubectl-opslevel"}, service.TagAssigns[0])
-	autopilot.Equals(t, 0, len(service.Tools))
-	autopilot.Equals(t, 0, len(service.Repositories))
-	// property assignment
-	autopilot.Equals(t, 5, len(service.Properties))
-	autopilot.Equals(t, "true", service.Properties["prop_bool"])
-	autopilot.Equals(t, "{}", service.Properties["prop_empty_object"])
-	autopilot.Equals(t, "", service.Properties["prop_empty_string"])
-	autopilot.Equals(t, `{"message":"hello world","condition":true}`, service.Properties["prop_object"])
-	autopilot.Equals(t, "hello world", service.Properties["prop_string"])
-}
 
-func TestJQServiceParserSampleConfig(t *testing.T) {
-	// Arrange
-	config, err := opslevel_jq_parser.NewServiceRegistrationConfig(opslevel_jq_parser.SampleConfig)
-	if err != nil {
-		t.Error(err)
-	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			config, err := opslevel_jq_parser.NewServiceRegistrationConfig(tc.config)
+			autopilot.Ok(t, err)
+			parser := opslevel_jq_parser.NewJQServiceParser(*config)
```

Comment:
Not a fan of this pointer unnecessary pointer deref so I am going to change that in a future PR.